src/test/java/org/broadinstitute/hellbender/tools/exome/segmentation/AlleleFractionSegmenterUnitTest.java

```diff
@@ -0,0 +1,87 @@
+package org.broadinstitute.hellbender.tools.exome.segmentation;
+
+import org.apache.commons.math3.distribution.BinomialDistribution;
+import org.apache.commons.math3.distribution.GammaDistribution;
+import org.apache.commons.math3.random.RandomGenerator;
+import org.apache.commons.math3.random.RandomGeneratorFactory;
+import org.broadinstitute.hellbender.tools.exome.ModeledSegment;
+import org.broadinstitute.hellbender.tools.exome.allelefraction.AllelicBiasParameters;
+import org.broadinstitute.hellbender.tools.exome.allelefraction.AllelicPanelOfNormals;
+import org.broadinstitute.hellbender.tools.exome.alleliccount.AllelicCount;
+import org.broadinstitute.hellbender.tools.exome.alleliccount.AllelicCountCollection;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Random;
+import java.util.stream.IntStream;
+
+/**
+ * Created by davidben on 5/16/16.
+ */
+public class AlleleFractionSegmenterUnitTest {
+    @Test
+    public void testSegmentation() {
+        final double[] trueWeights = new double[] {0.2, 0.5, 0.3};
+        final double[] trueMinorAlleleFractions = new double[] {0.12, 0.32, 0.5};
+        final double trueMemoryLength = 1e5;
+
+        final AllelicBiasParameters trueParams = new AllelicBiasParameters(1.0, 0.01, 0.01);
+
+        AlleleFractionHiddenMarkovModel trueModel = new AlleleFractionHiddenMarkovModel(trueMinorAlleleFractions, trueWeights,
+                trueMemoryLength, AllelicPanelOfNormals.EMPTY_PON, trueParams);
+
+        // randomly set positions
+        final Random random = new Random(271);
+        final int chainLength = 10000;
+        final List<SimpleInterval> positions = new ArrayList<>();
+        int position = 1;
+        for (int n = 0; n < chainLength; n++) {
+            position += random.nextInt((int) (trueMemoryLength/4));
+            final SimpleInterval interval = new SimpleInterval("chr1", position, position);
+            positions.add(interval);
+        }
+
+        final List<Integer> trueStates = trueModel.generateHiddenStateChain(positions);
+
+        //translate to ApacheCommons' parametrization of the gamma distribution
+        final double gammaShape = trueParams.getMeanBias() * trueParams.getMeanBias() / trueParams.getBiasVariance();
+        final double gammaScale = trueParams.getBiasVariance() / trueParams.getMeanBias();
+        final RandomGenerator rng = RandomGeneratorFactory.createRandomGenerator(new Random(563));
+        final GammaDistribution biasGenerator = new GammaDistribution(rng, gammaShape, gammaScale);
+
+        final AllelicCountCollection counts = new AllelicCountCollection();
```

Comment:
there's some code duplication here from `AlleleFractionSimulatedData`.  not sure if it's worth extracting (by making `AlleleFractionSimulatedData` a little more flexible) at this point, but i think it would be nice to eventually have proper classes that can generate fully specified simulated data for both allele fraction and copy ratio.